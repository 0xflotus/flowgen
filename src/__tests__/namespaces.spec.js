// @flow

import { compiler, beautify } from "..";

it("should handle namespaces", () => {
  const ts = `
namespace test {
  export const ok: number
}
`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should handle namespace merging", () => {
  const ts = `
namespace test {
  export const ok: number
}
namespace test {
  export const error: string
}
`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});

it("should handle namespace function merging", () => {
  const ts = `
namespace test {
  declare function test(err: number): void
}
namespace test {
  declare function test(response: string): string
}
`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
