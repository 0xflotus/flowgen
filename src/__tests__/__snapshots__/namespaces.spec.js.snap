// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle namespace function merging 1`] = `
"declare var npm$namespace$test: {
  test: typeof test$test
};
declare function test$test(err: number): void;

declare function test$test(response: string): string;
"
`;

exports[`should handle namespace merging 1`] = `
"declare var npm$namespace$test: {
  ok: typeof test$ok,
  error: typeof test$error
};
declare export var test$ok: number;

declare export var test$error: string;
"
`;

exports[`should handle namespaces 1`] = `
"declare var npm$namespace$test: {
  ok: typeof test$ok
};
declare export var test$ok: number;
"
`;
